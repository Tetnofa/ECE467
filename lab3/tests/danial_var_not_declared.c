//int z;
int main() { 
    // int x;
    // int z;
    int f;
    //g;
    //int y;
    {
        int z;
        {
            {
                {
                    f;
                    z;
                }
            }
        }
    }
}
void fcn () {
    int y;
    int z;
    int z;
    g;
}
int f;
void poo () {
    f;
}
int z;
int y;
int f;
bool fa () {
    int y;
    int z;
    int g;
    {
        int a;
        {
            a;
        }
    }
}