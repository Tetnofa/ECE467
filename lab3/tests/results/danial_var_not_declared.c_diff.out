0a1,65
> Program [useIO=0](2,0)
> 	Function[isProto=0]	Int(2,0)
> 		Identifier[name:main](2,4)
> 		Scope(2,11)
> 			Scalar Declaration			Int(5,4)
> 				Identifier[name:f](5,8)
> 			Scope(8,4)
> 				Scalar Declaration				Int(9,8)
> 					Identifier[name:z](9,12)
> 				Scope(10,8)
> 					Scope(11,12)
> 						Scope(12,16)
> 							Expression Statement(13,20)
> 								Int Expression(13,20)
> 									Reference(13,20)
> 										Identifier[name:f](13,20)
> 							Expression Statement(14,20)
> 								Int Expression(14,20)
> 									Reference(14,20)
> 										Identifier[name:z](14,20)
> 	Function[isProto=0]	Void(20,0)
> 		Identifier[name:fcn](20,5)
> 		Scope(20,12)
> 			Scalar Declaration			Int(21,4)
> 				Identifier[name:y](21,8)
> 			Scalar Declaration			Int(22,4)
> 				Identifier[name:z](22,8)
> 			Scalar Declaration			Int(23,4)
> 				Identifier[name:z](23,8)
> 			Expression Statement(24,4)
> 				Int Expression(24,4)
> 					Reference(24,4)
> 						Identifier[name:g](24,4)
> 	Scalar Declaration	Int(26,0)
> 		Identifier[name:f](26,4)
> 	Function[isProto=0]	Void(27,0)
> 		Identifier[name:poo](27,5)
> 		Scope(27,12)
> 			Expression Statement(28,4)
> 				Int Expression(28,4)
> 					Reference(28,4)
> 						Identifier[name:f](28,4)
> 	Scalar Declaration	Int(30,0)
> 		Identifier[name:z](30,4)
> 	Scalar Declaration	Int(31,0)
> 		Identifier[name:y](31,4)
> 	Scalar Declaration	Int(32,0)
> 		Identifier[name:f](32,4)
> 	Function[isProto=0]	Bool(33,0)
> 		Identifier[name:fa](33,5)
> 		Scope(33,11)
> 			Scalar Declaration			Int(34,4)
> 				Identifier[name:y](34,8)
> 			Scalar Declaration			Int(35,4)
> 				Identifier[name:z](35,8)
> 			Scalar Declaration			Int(36,4)
> 				Identifier[name:g](36,8)
> 			Scope(37,4)
> 				Scalar Declaration				Int(38,8)
> 					Identifier[name:a](38,12)
> 				Scope(39,8)
> 					Expression Statement(40,12)
> 						Int Expression(40,12)
> 							Reference(40,12)
> 								Identifier[name:a](40,12)
